### ISSUE TITLE
Jupyterhub not accessible using Azure Cloud

### ISSUE THREAD

**Issue by not4everybody** [jupyterhub/jupyterhub#2304]
Hey,

I have two Azure cloud machines running and trying to use jupyterhub on them. One is a 'standard' ubuntu server, and one is a machine learning version where jupyterhub is already installed and running.
With both I encounter the same issue that I cannot access/use the jupyterhub. Every time I try to go to the website via a browser I get a time out.
I got the network setting set to allow incoming traffic via http, https, ssh. I tried to access it from my phone (using my LTE network not my WI-FI), yet nothing seems to give me access. All I get is a blank page...
On the 'standard machine' I instantly get a 'connection refused' error, on the 'machine learning vm' after a looooooong time I get: 'internal error - server connection terminated'

Does any one know, what could possibly be the problem here? I'm googling my way around for the last two hours but cannot find a solution....

Thanks,
Anja

**Comment by minrk** [comment]
Both of these sound like firewall/network configurations of the cloud provider not allowing access. I guess the thing to check are:

1. is jupyterhub listening on the port you have exposed?
2. can you run any other lightweight servers (e.g. `python -m http.server`) to verify that there isn't something specific about jupyterhub that's failing to be connectable

**Comment by not4everybody** [comment]
Yes, I usually have a python flask application runing on the 'standard machine'. I need to set the ip to '0.0.0.0' and port=80 in order to run flask properly, but that is working like a charm.

I tried to run jupyterhub using --ip=0.0.0.0 --port=80 but then I get an error in the console saying TimeoutError: Server didn't respond in 1 seconds.
I also tried to add c.JupyterHub.hub_ip with the adress of my server and get the same error in the console.
It seems as if jupyterhub only 'correctly' starts up with the standard arguments but then I cannot access it in a browser, as soon as I try to change the arguments jupyterhub doesn't start up at all...

**Comment by minrk** [comment]
hub_ip is the private IP of the hub. That shouldn't be changed. The default behavior of jupyterhub is to listen on all interfaces, so just setting `--port=80` ought to be enough.

If changing the arguments to jupyterhub causes it to not start, can you share the logs of the failure? Is there perhaps something else already on port 80 or a permission issue binding to the privileged port?
